### ISSUE TITLE
antiklepto / anti-exfil protocol for Schnorr sigs

### ISSUE THREAD

**Issue by benma** [ElementsProject/secp256k1-zkp#172]
https://github.com/ElementsProject/secp256k1-zkp/blob/725d895fc54cf82da1c2a9c69048656405da556d/include/secp256k1_ecdsa_s2c.h#L100

Is anyone working on implementing this protocol for Schnorr signatures? Since Taproot is now active, it would be great to have this available to implement in hardware wallets. 

@jonasnick @apoelstra

**Comment by jonasnick** [comment]
All I know is that @dr-orlovsky opened issue #154 for adding sign-to-contract and opened PR https://github.com/bitcoin-core/secp256k1/pull/1018 with an implementaion.

**Comment by real-or-random** [comment]
What about https://github.com/bitcoin-core/secp256k1/pull/590 ?

**Comment by dr-orlovsky** [comment]
Yes, I had a plan to work on that, but it could take more time than I originally expected - got really overflow by all the problems during the last months since the war began.

Still pan to work on it one the load will decrease - but if there will be anyone willing to work on it now - I will not object.

**Comment by benma** [comment]
https://github.com/ElementsProject/secp256k1-zkp/issues/154 is a prerequisite for antiklepto - I will work on it.

**Comment by benma** [comment]
I opened a draft implementation here: https://github.com/bitcoin-core/secp256k1/pull/1140
